rust/libnewsboat/src/cliargsparser.rs

```diff
@@ -179,6 +188,18 @@ pub fn parse_cliargs(opts: Vec<OsString>, args: &mut CliArgsParser) -> Result<()
                 let config_file = parser.value()?;
                 args.config_file = resolve_path(&config_file);
             }
+            Long("queue-file") => {
+                let queue_file = parser.value()?;
+                args.queue_file = resolve_path(&queue_file);
+            }
+            Long("history-search") => {
+                let search_file = parser.value()?;
+                args.search_file = resolve_path(&search_file);
+            }
+            Long("history-cmdline") => {
+                let cmdline_file = parser.value()?;
+                args.cmdline_file = resolve_path(&cmdline_file);
+            }
```

Comment:
I realize you named the options after the files, but I think it's better to flip the order because it reads more like English: "search history", "cmdline history". Also, add "file" because all other options that take files have it in their names. So, please rename this to `search-history-file` and `cmdline-history-file`.